app/assets/javascripts/app/views/single-post-viewer/single_post_content_view.js

```diff
@@ -37,6 +37,42 @@ app.views.SinglePostContent = app.views.Base.extend({
 
   showPost : function() {
     return (app.currentUser.get("showNsfw")) || !this.model.get("nsfw");
+  },
+
+  map: function() {
+    var coordinates = this.model.get("coordinates");
+
+    // if (coordinates != "" && tileserver.enable) {  // for when the tileserver is set via the diaspora.yml
+    if (coordinates.lat) {
+      $( ".map" ).append( "<div id='map'></div>" ); // this needs a new subview?
+      var tileLayerSource = "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}";
+      var map = L.map("map").setView([coordinates.lat, coordinates.lng], 16);
+      var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
+                        "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                        "Imagery © <a href='http://mapbox.com'>Mapbox</a>";
+
+      L.tileLayer(tileLayerSource, {
+        attribution:  attribution,
+        maxZoom: 18,
+        id: "zaziemo.mpn66kn8",
+        accessToken: "pk.eyJ1IjoiemF6aWVtbyIsImEiOiI3ODVjMzVjNmM2ZTU3YWM3YTE5YWYwMTRhODljM2M1MSJ9.-nVgyS4PLnV4m9YkvMB5wA"
+      }).addTo(map);
+
+      var marker = L.icon({
+          iconUrl: ImagePaths.get("leaflet-marker-icon.png"),
+          iconRetinaUrl: ImagePaths.get("leaflet-marker-icon-2x.png"),
+          shadowUrl: ImagePaths.get("leaflet-marker-shadow.png"),
+          shadowRetinaUrl: ImagePaths.get("leaflet-marker-shadow.png"),
+      });
+
+      var markerOnMap = L.marker(coordinates, {icon: marker}).addTo(map);
+
+      return map;
+    }
+  },
+
+  postRenderTemplate : function(){
+    _.defer(_.bind(this.map, this));
```

Comment:
@jaywink yes, the map will open up automatically in the single post view. In the stream it is not shown automatically but a click on the location leads to the single post view.
@Flaburgan it is no problem to have it placed before the polls or make it smaller. Inside the head it could at most have a thumbnail size … but we can try that as well.